### PR TITLE
Adds a new type of material stack: insulated fiber

### PR DESCRIPTION
**Pull request by ClownMoff**
## About The Pull Request

Allows to get insulated fibers, a new material you get from cutting insualated gloves and budget insulated gloves with a knife. You get 5 fibers from insulated gloves and 1 and 4 cloth from budget insulated gloves.

## Why It's Good For The Game

You can get a pair of insulated gloves if you get 5 budget insulated gloves, which allows for an alternative way of getting that item. Getting insulated fibers from insulated gloves doesn't make much sense now (since they are only used to craft insuls), but in the near future it will be an ingredient for a beer brewing PR i'm making and it doesn't make sense making this change in that PR.
 
## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

imagen
imagen
imagen
imagen
imagen

</details>

## Changelog
:cl: ClownMoff
add: Adds a new material called insulated fiber, found in insulated and budget insulated gloves by cutting them with a knife. You can craft insulated gloves with 5 insulated fibers
/:cl:

### CODE REVIEW

**Comment by ClownMoff** [comment]
Why a knife? Because wire cutters are already reserved for cutting the tips off the fingers.

**Comment by Tsar-Salat** [comment]
> BUT, you can get a pair of insulated gloves if you get 5 budget insulated gloves, which allows for an alternative way of getting that item.

This should be your primary reason. Not a BUT at the end of the reasoning.



> ## Changelog
> 🆑 ClownMoff add: Adds a new material called insulated fiber, found in insulated and budget insulated gloves by cutting them with a knife /:cl:

You failed to mention that you can now craft insults. That is what players will care about.

**Comment by Fronsis** [comment]
Hmmm... In the past we had a limitation of how to acquire insuls due to greyshirts going bananas, this however doesn't seem that bad, like.. if anything i can only see only one or two person getting insuls since the budget insul gloves are not available in mass quantities  in maintenance and they're subject to RNG so i think it's a nice alternative that requires some exploring and gathering

File: code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm [hunk 1]
```diff
@@ -139,3 +139,23 @@ GLOBAL_LIST_INIT(sandbag_recipes, list ( \
...
+	singular_name = "insulated fiber"
+	icon_state = "insulated_fibers"
+	icon = 'icons/obj/stacks/miscellaneous.dmi'
+	item_state = "insulated_fibers"
+	resistance_flags = FLAMMABLE
+	force = 0
+	throwforce = 0
+	merge_type = /obj/item/stack/sheet/insulated_fibers
+	var/pull_effort = 30
+	var/loom_result = /obj/item/stack/sheet/insulated_fibers
```
Comment (DrDuckedGoose) [inline code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm:157]: What are these variables for? It doesn't seem like they're used elsewhere.

File: code/modules/clothing/gloves/color.dm [hunk 2]
```diff
@@ -40,6 +40,19 @@
...
+		var/obj/item/stack/sheet/insulated_fibers/F = new (T, 5)
+		if(QDELETED(F))
+			F = locate(/obj/item/stack/sheet/insulated_fibers/) in T
+		if(F)
+			transfer_fingerprints_to(F)
+			F.add_fingerprint(user)
+		qdel(src)
+		to_chat(user, "<span class='notice'>You tear [src] apart.</span>")
+	else
+		return ..()
```
Comment (DrDuckedGoose) [inline code/modules/clothing/gloves/color.dm:55]: This and the code for ``/obj/item/clothing/gloves/color/fyellow`` is very similar. Do you think you could try minimizing the code duplication?

File: code/modules/clothing/gloves/color.dm [hunk 3]
```diff
@@ -64,6 +77,26 @@
...
+		if(QDELETED(C))
+			C = locate(/obj/item/stack/sheet/insulated_fibers/) in T
+		if(C)
+			transfer_fingerprints_to(C)
+			C.add_fingerprint(user)
+		qdel(src)
+		to_chat(user, "<span class='notice'>You tear [src] apart.</span>")
+	else
+		return ..()
+
```
Comment (DrDuckedGoose) [inline code/modules/clothing/gloves/color.dm:99]: Slight code duplication, looks like it could be minimized.

**Review by DrDuckedGoose — CHANGES_REQUESTED** [review]
Took a quick look over it 👍

File: code/modules/clothing/gloves/color.dm (hunk 2 above)
Comment (ClownMoff) [reply code/modules/clothing/gloves/color.dm:55]: How? Do I make it a subtype of the insulated ones? I'm still pretty new on coding so I don't know how to clean code to not be duplicated like this

Comment (ToasterBan) [reply code/modules/clothing/gloves/color.dm:55]: could have cutting gloves be a thing on all gloves, and give insulated and fake insulated gloves variables to give their respected drops

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
Use a personal crafting recipe, not the material recipe.

**Comment by ClownMoff** [comment]
What do you mean? Using the crafting menu to craft It instead of the fibers itself?

**Comment by EvilDragonfiend** [comment]
> What do you mean? Using the crafting menu to craft It instead of the fibers itself?

Yes

File: code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm (hunk 1 above)
Comment (ClownMoff) [reply code/game/objects/items/stacks/sheets/miscellaneous/miscellaneous_mats.dm:157]: idk I copypasted

**Comment by EvilDragonfiend** [comment]
Well, just do as you'd like. It's better to be not abandoned, I guess.

**Comment by Nakalon** [comment]
I kindly asked varo to incorporate this into my material scrap and salvaging system introduced in #11494

As such, I will pick this idea up and introducing it in a later PR

**Pull request closed without merging**